### PR TITLE
 fix: correctly generate numeric unicode subset faces [#949]

### LINKED ISSUE #949: Chinese Noto fonts mixin does not allow using symbolic subsets
### Describe the bug

I am using the `faces` mixin to integrate fonts in my SASS files, and due to some weird defaults (most fonts will only produce CSS for the `latin` subset!) I need to specify the font subsets to use.

This  worked well for most Noto fonts, but for the Chinese fonts it did not generate any CSS.

### Steps to Reproduce

@use '@fontsource-variable/noto-sans-sc/scss/mixins' as NotoSansSC
@include NotoSerifSC.faces($directory: $fonts_dir, $subsets: all)

### Expected behavior

Multiple `@font-face`definitions for the NotoSansSC font. But nothing is generated

### Version

_No response_

### OS

_No response_

### Browser

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by RBilly**
Template from `packages/cli/src/sass/mixins.ts` file make use of the `metada` file generated by `packages/cli/src/sass/metada.ts`. In this `metada` file, each subset of a font is represented by a unicode subset defined by a unicode range but sometime, the main font subset can be represented by numeric subsets. 
When calling the mixin function with this `main` subset specified as parameter, there is an issue where the mixin function tries to verify that this `main` subset is equal to the `default` subset. But this is rarely the case because most of the time the `default` subset is almost always set to `latin`, so it fails to produce the expected output.

Change made in `packages/cli/src/sass/mixins.ts` check that if the the subset value passed as parameter does not have a unicode subset but is part of the list of the font subsets then we use the numeric unicode subsets.

Test case was update to be able to pass options to the mixin call and catch this case as well.

side note: indentation of some files were also modified but it does not seem like an error since they seem to be aligned with the content of the editorconfig file.

### CODE REVIEW

File: packages/cli/src/sass/mixins.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ $displayVar: null !default;
         ($subset == $unicodeSubset) or
           (
             // Is numeric subset
```
Comment (ayuhito) [inline packages/cli/src/sass/mixins.ts]: Could you update this comment to explain what's going on?

**Review by ayuhito — CHANGES_REQUESTED** [review]
I think everything looks good. This shouldn't be breaking, right?

File: packages/cli/src/sass/mixins.ts (hunk 1 above)
Comment (RBilly) [reply packages/cli/src/sass/mixins.ts]: Done, and it also made me realize that I missed some use-case. I added some more test to cover that as well

**Comment by RBilly** [comment]
> I think everything looks good. This shouldn't be breaking, right?

I do not consider this a breaking change but there is some impact.
Previously, when calling the faces mixin without a subsets parameter, it would use the default subset instead. So it would generate the numeric unicode faces + latin. 
With my change I could either let the default subset be used but then it would only generate "latin" faces when no subsets were provided which does not make sense for a lot of fonts that have latin as default. Or I could just say that if no subsets are provided, it means all subsets should be used. 
I chose the later but it means that for some user they could import more faces than before which is more correct but it could unexpectedly increase their bundle size.

**Review by ayuhito — APPROVED** [review]

**Comment by ayuhito** [comment]
> I chose the later but it means that for some user they could import more faces than before which is more correct but it could unexpectedly increase their bundle size.

Widening the subset range is fine since users won't necessarily download the extra font files unless their webpage uses unicode characters outside the latin subset.

Thank you very much for your contribution! I think it looks good to me! 🎉

**Pull request merged**